src/main/Config.h

```diff
@@ -499,6 +499,13 @@ class Config : public std::enable_shared_from_this<Config>
     // The default value is false.
     bool HALT_ON_INTERNAL_TRANSACTION_ERROR;
 
+    // If set to true, env will return additional diagnostics events that are
+    // not part of the protocol. These events will be put into a list in the
+    // non-hashed portion of the meta, and this list will contain all events
+    // so ordering can be maintained between all events.
+    // The default value is false, and this should not be enabled on validators.
+    bool ENABLE_SOROBAN_DIAGNOSTIC_EVENTS;
```

Comment:
Please also add this to the example config and the docs.

Reply:
I added this to the example config, and also put the setting around `ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION` guards.

Reply:
Ah, sorry, I actually forgot that the setting may not be compiled in... Maybe in this case we don't need to include it in cfg to not confuse users? (I think it depends on whether the core crashes on it if the core with the current protocol is built). Sorry for the confusion.

Reply:
I was actually thinking about this when you first commented. 
1. I thought it would be better if we included it in the current config.
2. Does it matter if core crashes if soroban isn't compiled in? Do we run tests using the example config anywhere?

Reply:
I'm not sure if the example config is being used directly. We probably can leave it to make sure we don't forget about it.